allennlp/training/deepspeed/sparse_transformer_embedder.py

```diff
@@ -0,0 +1,11 @@
+from allennlp.modules.token_embedders.token_embedder import TokenEmbedder
+from allennlp.modules.token_embedders.pretrained_transformer_embedder import PretrainedTransformerEmbedder
+
+from deepspeed.ops.sparse_attention.sparse_attention_utils import SparseAttentionUtils
+
+# Doesn't work yet
```

Comment:
same

Reply:
This is just a question of scope. I think it would be a very neat little addition for not much cost, but I've been having trouble getting it installed properly (doing all of this on a SLURM cluster is leading to most of these problems, esp. not being able to use Docker). Fwiw, someone in the issue thread was expressing interest in this.

So overall it's just a question of if you'd like me to give it a shot to get it working now, or leave it for the future. Happy either way.

Reply:
I think it would be a great addition, but let's not make this PR bigger than it already is. It would be a nice follow-up though, especially if those classes end up being the secret to those massive promises that they have on their website.

You can make PRs that branch off of other PRs, just in case you didn't know :-P